testcafe/pages/ServicePage.js

```diff
@@ -20,4 +20,17 @@ export default class ServicePage {
     this.schedule = baseSelector.findReact('TableOfOpeningTimes tbody tr');
     this.url = serviceId => `${config.baseUrl}/services/${serviceId}`;
   }
+
+  /**
+   * Wait until the page is fully loaded.
+   *
+   * Some TestCafe actions may attempt to run before a page is fully loaded, so
+   * this method can be called to force us to wait until the page is loaded
+   * first.
+   *
+   * @param t - A TestCafe test Promise.
+   */
+  async waitUntilPageLoaded(t) {
+    await t.hover(this.name);
```

Comment:
So much better

Reply:
I think we use `t.hover` in a few other test modules but we can update that as needed

Reply:
Yeah, we can update them as we go. We also might want to upgrade the version of TestCafe that we're using, since I think they have some of the bugs with not properly waiting for elements to appear before acting on them fixed.